src/arviz_plots/plots/mcseplot.py

```diff
@@ -0,0 +1,491 @@
+            for var_name, sel, isel in plotters:
+                print(f"\nvar={var_name} | sel={sel} | sel.items()={sel.items()} | isel={isel}")
+                da = distribution[var_name].sel(sel)
+                da = da.values.flatten()
+                # print(f"\n da = {da}")
+                quantile_values = array_stats.quantile(da, probs)
+                # print(f"quantile_values = {quantile_values}")
+
+                # Convert quantile_values to a DataArray
+                quantile_da = xr.DataArray(quantile_values, dims=["mcse_dim"], name=var_name)
+                print(f"\n quantile_da = {quantile_da}")
+
+                # Expand dims of dataarray and assign new coords 'z' and sel dim coords
+                quantile_da_expanded = quantile_da.expand_dims({"plot_axis": ["z"]})
+                # .assign_coords(
+                #    {"plot_axis": ["z"]}
+                # )
+                # .to_dataset(  # .assign_coords(plot_axis=["z"])
+                #    name=var_name
+                # )
+                if sel:  # adding sel dims to quantile_da
+                    sel_expanded = {key: [value] for key, value in sel.items()}
+                    print(f"sel_expanded = {sel_expanded}")
+                    quantile_da_expanded = quantile_da_expanded.expand_dims(sel_expanded)
+                    #    dim=list(sel.keys())
+                    # ).assign_coords(**sel)
+                # quantile_da_expanded = quantile_da_expanded.assign_coords(**sel)
+
+                print(f"\n quantile_da_expanded = {quantile_da_expanded}")
+
+                # print(f"\n mcse_dataset sel = {mcse_dataset[var_name].sel(**sel)!r}")
+                # print(f"\n mcse_dataset isel = {mcse_dataset[var_name].isel(isel)!r}")
+
+                if var_name in z_mcse_dataset:
+                    combined_dataarray = z_mcse_dataset[var_name].combine_first(
+                        quantile_da_expanded
+                    )
+                    print(f"\n combined_dataarray = {combined_dataarray}")
+                    z_mcse_dataset.update({var_name: (var_name, combined_dataarray)})
+                    # .concat(), .merge() are not in-place functions. .update() is in-place but
+                    # not working rn
+
+                    # z_mcse_dataset[var_name] =
+                    # xr.concat(
+                    #    [z_mcse_dataset[var_name], quantile_da_expanded],
+                    #    dim="school",
+                    # )
+                else:
+                    z_mcse_dataset[var_name] = quantile_da_expanded
+                    # print(f"\n adding var {var_name} to z_mcse_dataset: {z_mcse_dataset}")
+
+            print(f"\n z_mcse_dataset = {z_mcse_dataset}")"""
+            print(len(a))
+
+            # print(f"\n final z_mcse_dataset = {z_mcse_dataset}")
+
+            # for now the quantile_values can be computed in the visual element function itself
```

Comment:
you already have `xdata_dataset`, also `mcse_y_dataset` (which now should be the errorbars, not the y position). I think `quantiles_dataset = distribution.quantile(probs, dim=mcse_dims)` will mostly give you what you want, the only problem is it will have a `quantile` dimension instead of an `mcse_dim` one, you can rename and reindex it or use quantile as dimension name for the other objects.

afer that you should be able to concat them into a new object with `plot_axis` having `x` (xdata_dataset), `y` (quantiles_dataset) and `error` (mcse_y_dataset) so the visual function can process that and still accept x, y in case we want aesthetics mapped to them